apps/tron_app/tron_helpers.c

```diff
@@ -0,0 +1,123 @@
+ * rights under the License will not include, and the License does not
+ * grant to you, the right to Sell the Software.
+ *
+ * For purposes of the foregoing, "Sell" means practicing any or all
+ * of the rights granted to you under the License to provide to third
+ * parties, for a fee or other consideration (including without
+ * limitation fees for hosting or consulting/ support services related
+ * to the Software), a product or service whose value derives, entirely
+ * or substantially, from the functionality of the Software. Any license
+ * notice or attribution required by the License must also include
+ * this Commons Clause License Condition notice.
+ *
+ * Software: All X1Wallet associated files.
+ * License: MIT
+ * Licensor: HODL TECH PTE LTD
+ *
+ ******************************************************************************
+ */
+
+/*****************************************************************************
+ * INCLUDES
+ *****************************************************************************/
+
+#include "tron_helpers.h"
+
+/*****************************************************************************
+ * EXTERN VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE MACROS AND DEFINES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE TYPEDEFS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTION PROTOTYPES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * GLOBAL VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTIONS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * GLOBAL FUNCTIONS
+ *****************************************************************************/
+
+bool tron_derivation_path_guard(const uint32_t *path, uint8_t levels) {
+  bool status = false;
+  if (levels < 3)
+    return status;
+
+  uint32_t purpose = path[0], coin = path[1], account = path[2];
+
+  switch (levels) {
+    case 3: {    // m/44'/195'/0'
+      status = (purpose == TRON_PURPOSE_INDEX && coin == TRON_COIN_INDEX &&
+                account == TRON_ACCOUNT_INDEX);
+    } break;
+
```

Comment:
Does case 4 exist? Include the reference for this.

Reply:
1st one (m/44'/195'/0') is tron's standard path, 
second one (m/44'/195'/0'/0/i) is popular path used by adopted by other wallets

Reply:
updated